### ISSUE TITLE
Strange behaviour in `Quantiles`

### ISSUE THREAD

**Issue by darribas** [pysal/mapclassify#2]
`Quantiles` seems to return numbers of bins that differ from `k` in some contexts:

```python
In [45]: y = db['HR60'] 
In [46]: breaks = ps.Quantiles(y, 9).bins 
In [47]: len(breaks) 
Out[47]: 8

In [49]: y.min()
Out[49]: 0.0

In [50]: y.max()
Out[50]: 92.936802974000003

In [51]: breaks
Out[51]:
array([  0.        ,   1.13094026,   2.19806302,   3.44429745,
         5.13821807,   7.47789373,  10.86644222,  92.93680297])

In [67]: [print(i, ps.Quantiles(y, k=i).bins.shape[0]) for i in range(1, 10)] 
1 1 
2 2 
3 3 
4 4 
5 5 
6 6 
7 7 
8 7 
9 8 
Out[67]:
```

**Comment by ljwolf** [comment]
hmm. I think this happens because the percentile breaks are not unique:

```python
>>> k = 8
>>> w = 100.0 / k
>>> ptiles = numpy.arange(w, 100+w, w)
>>> np.percentile(df.HR60, ptiles)
array([  0.        ,   0.        ,   1.53881101,   2.78332994,
         4.44488918,   6.88530147,  10.32196185,  92.93680297])
```
the `mapclassify.classifiers.quantile` function takes these quantiles and calls `np.unique` on them, dropping a class. 

Not sure what the "right" thing to do here is if two classes are effectively the same quantile. 

![aout](https://cloud.githubusercontent.com/assets/2250995/25864158/67d67f36-34bc-11e7-9521-e7e20eaf1acc.png)

**Comment by sjsrey** [comment]
That is the intended [behavior:]( https://github.com/pysal/mapclassify/blob/master/mapclassify/classifiers.py#L78)

```
   Examples
    --------

    >>> x = [1.0] * 100
    >>> x.extend([3.0] * 40)
    >>> len(x)
    140
    >>> y = np.array(x)
    >>> quantile(y)
    array([ 1.,  3.])

    Note that if there are enough ties that the quantile values repeat, we
    collapse to pseudo quantiles in which case the number of classes will be
    less than k
```

**Comment by ljwolf** [comment]
resolved by @sjsrey in #3
